src/main/java/com/impactupgrade/nucleus/client/SfdcClient.java

```diff
@@ -646,6 +646,92 @@ protected QueryResult querySmsContacts(String updatedSinceClause, String filter,
+      env.logJobWarn("no filter provided; out of caution, skipping the query to protect API limits");
+      return new QueryResult();
+    }
+    String updatedContactsQuery = "SELECT Id FROM Contact " +
+        "WHERE " + updatedSinceClause;
+    List<SObject> updatedContacts = queryListAutoPaged(updatedContactsQuery);
+    if (CollectionUtils.isEmpty(updatedContacts)) {
+      return new QueryResult();
+    }
+    String updatedContactIds = updatedContacts.stream()
+        .map(sObject -> "'" + sObject.getField("Id") + "'")
+        .collect(Collectors.joining(","));
+
+    String totalOpportunityAmountQuery = "SELECT ContactId, SUM(Amount) TotalAmount FROM Opportunity " +
+        "WHERE ContactId IN (" + updatedContactIds + ")" +
+        "GROUP BY ContactId " +
+        "HAVING SUM(Amount) > 0";
+    List<SObject> aggregateResults = queryListAutoPaged(totalOpportunityAmountQuery);
+    if (aggregateResults.isEmpty()) {
+      return new QueryResult();
+    }
+
+    String contactIds = aggregateResults.stream()
+        .map(sObject -> "'" + sObject.getField("ContactId") + "'")
+        .collect(Collectors.joining(","));
+    String query = "SELECT " + getFieldsList(CONTACT_FIELDS, env.getConfig().salesforce.customQueryFields.contact, null) + " " +
+        "FROM Contact " +
+        "WHERE Id IN (" + contactIds + ")";
+    return query(query);
+  }
+
+  protected QueryResult queryDonatingAccountsContacts(String updatedSinceClause) throws ConnectionException, InterruptedException {
+    if (Strings.isNullOrEmpty(updatedSinceClause)) {
+      env.logJobWarn("no filter provided; out of caution, skipping the query to protect API limits");
+      return new QueryResult();
+    }
+    String updatedAccountsQuery = "SELECT Id FROM Account " +
+        "WHERE " + updatedSinceClause;
+    List<SObject> updatedAccounts = queryListAutoPaged(updatedAccountsQuery);
+    if (CollectionUtils.isEmpty(updatedAccounts)) {
+      return new QueryResult();
+    }
+    String updatedAccountIds = updatedAccounts.stream()
+        .map(sObject -> "'" + sObject.getField("Id") + "'")
+        .collect(Collectors.joining(","));
+
+    String totalOpportunityAmountQuery = "SELECT AccountId, SUM(Amount) TotalAmount FROM Opportunity " +
+        "WHERE AccountId IN (" + updatedAccountIds + ")" +
+        "GROUP BY AccountId " +
+        "HAVING SUM(Amount) > 0";
+    List<SObject> aggregateResults = queryListAutoPaged(totalOpportunityAmountQuery);
+    if (aggregateResults.isEmpty()) {
+      return new QueryResult();
+    }
+
+    String accountIds = aggregateResults.stream()
+        .map(sObject -> "'" + sObject.getField("AccountId") + "'")
+        .collect(Collectors.joining(","));
+    String query = "SELECT " + getFieldsList(CONTACT_FIELDS, env.getConfig().salesforce.customQueryFields.contact, null) + " " +
+        "FROM Contact " +
+        "WHERE AccountId IN (" + accountIds + ")";
+    return query(query);
```

Comment:
All the same points above about the donor contacts -- consolidate into a single query?